esx_service/utils/docker-volume-auth-vmodl.java

```diff
@@ -0,0 +1,96 @@
+/*
+# Copyright 2016 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+*/
+
+import vmodl.*;
+
+/**
+ *
+ * DockerVolumeAuthManager is the top-level managed object that provides APIs to manage docker volume
+ *  authorization objects such as tenants and datastore access privileges.
+ *
+ * <p>
+ * <b>Tenants</b> are a group of VMs that are granted privileges to create, delete and mount
+ * docker volumes (VMDKs) on one or more datastores. Tenants provide for full isolation of these
+ * volumes such that one tenant cannot see or manipulate volumes of another tenant even if both
+ * tenants have volumes residing on the same datastore. The sole exception to this rule is if the
+ * global_visibility privilege is assigned to a given datastore. In that case the tenant has all the
+ * other privileges granted for all volumes in the datastore, not just ones it created.
+ * <p>
+ * <b>DatastoreAccessPrivileges</b> are predefined operations against a datastore and limits on those
+ * operations.
+ */
+@managed public interface DockerVolumeAuthManager {
+
+   /* An array of references to Tenant managed entities */
+   @readonly @optional Tenant[] tenants();
+
+   /* VMs that are not members of a tenant */
+   @readonly @optional VirtualMachine[] availableVMs();
+
+   @task Tenant createTenant(
+         String name,
+         String description,
+         VirtualMachine[] vms,
+         DatastoreAccessPrivileges[] privileges)
+   throws AlreadyExists;
+
+   @task Tenant removeTenant(String id, Boolean deleteVolumes);
+   @task Tenant[] listTenants();
+   /* TBD: which infomation need to be returned by this function
```

Comment:
please drop this comment. Also please add to the top of the file "*** NOTE: this is  draft, the actual model will be done via Python extension"